### ISSUE TITLE
Autocomplete in 1.4-SNAPSHOT builds endpoint URL incorrectly, unusable in Lift 3.0-M5

### ISSUE THREAD

**Issue by bgrace** [liftmodules/widgets#20]
The autocomplete widget attempts to fire Ajax at URLs of the following form: `http://lift/ajax/...etc.` rather than creating a relative URL.

The [line referenced](https://github.com/liftmodules/widgets/blob/master/src/main/scala/net/liftmodules/widgets/autocomplete/AutoComplete.scala#L181) in issue #19 is the problem:

```
val what: String = encodeURL(S.contextPath + "/" + LiftRules.liftPath+"/ajax/"+S.renderVersion+"?"+func+"=foo")
```

The definition of [`LiftRules.liftPath`](https://github.com/lift/framework/blob/3.0-M5-release/web/webkit/src/main/scala/net/liftweb/http/LiftRules.scala#L1204) is:

`def liftPath: String = S.contextPath + "/" + liftContextRelativePath`

So the problem is clear, `S.contextPath + "/"` is prepended twice; and if `S.contextPath` is the empty string the result is `//`.

For me, the following replacement works:
`val what: String = encodeURL(LiftRules.liftPath + "/ajax/"+ S.renderVersion+"?"+func+"=foo")`

However I'm not well-versed enough in LiftRules to be confident it's going to play well in all cases.

**Comment by Shadowfiend** [comment]
I think this is the correct solution, yes.

**Comment by erysavy** [comment]
I just encountered the same issue. Having the application running in root context, autocompletion doesn't work because //lift/ajax/... is translated by XHR to http://lift/ajax/...

Applying the modification mentioned above solves the problem, thanks!

This module works great, isn't anybody else using it? I would greatly appreciate this being fixed in the official repository as well.

**Comment by Shadowfiend** [comment]
If you issue a pull request with the fix in question, I imagine it'll be merged pretty quickly. If I have merge rights I'll probably let it sit for a bit in case anyone has any comments and then merge it.

**Comment by erysavy** [comment]
Sorry, my error, I wanted to add the autocomplete fix, but it looks I don't have push permissions - ?

**Comment by d6y** [comment]
Thank you @erysavy for the fix.  This is now available as 1.4.1 for Lift 3 (Scala 2.11 and 2.12).
